src/hotspot/share/jvmci/jvmciCompilerToVM.cpp

```diff
@@ -728,7 +728,7 @@ C2V_END
 
 C2V_VMENTRY_0(jint, getVtableIndexForInterfaceMethod, (JNIEnv* env, jobject, jobject jvmci_type, jobject jvmci_method))
   Klass* klass = JVMCIENV->asKlass(jvmci_type);
-  methodHandle method(THREAD, JVMCIENV->asMethod(jvmci_method));
+  Method* method = JVMCIENV->asMethod(jvmci_method);
```

Comment:
The reason that resolved_method was a methodHandle is in case of redefinition, we need to know if code is referring to this version of the method so that it's not deallocated.   It's enough for one of the callers to create a methodHandle but passing the methodHandle will guarantee it.  I'm not sure why you needed to make this change.

Reply:
I refactored `getVtableIndexForInterfaceMethod` and kept the handle in place while dereferencing it when passing into `vtable_index_of_interface_method`. Does it look better now?

Reply:
Ok, this is fine.  Be careful if you have new callers for this function.